### ISSUE TITLE
Change the way restrictinfra works

### ISSUE THREAD

**Issue by pranavtifr** [longturn/freeciv21#576]
**Is your feature request related to a problem? Please describe.**
Shifting borders sometimes cause attacks to be incredibly swift where people lose entire kingdoms in a single turn. 

**Describe the solution you'd like**
One prevailing idea is to have restrictinfra as an extension of ZOC, where each unit has a strength for imposing restrictinfra around it. Similarly own borders has a strength and also, cities. Final restrictinfra is an accumulation of all these strengths. 

**Describe alternatives you've considered**
Other ideas for the problem were changing borders to have something called noman's land. Changing borders only at tc, or  restrictinfra around newly conquered city irrespective of borders.

**Comment by jwrober** [comment]
Oh I thought I fixed my typo on the incorrectly tagged PR. Thanks

**Comment by alain-bkr** [comment]
I agree restrictinfra could be inmproved : current 0 / 1 is to crude.
Maybe an option with a percentage , 
 - 0 = no restrictinfra
 - 100% = actual restrict infra = only 1 tile move
 - and for example 50% = halves the movements points of the unit ?
 
I don't know if this is simpler than the proposed idea.
